config/theme.go

```diff
@@ -490,6 +530,13 @@ type ThemeExplain struct {
 	Required color.Color `defaultFrom:"theme.base.danger"` // used on the trailing "-required-" string
 }
 
+// ThemeDiff holds colors for the "kubectl diff" output.
+type ThemeDiff struct {
+	Added     color.Color `defaultFrom:"theme.base.added"`     // used on added lines
+	Removed   color.Color `defaultFrom:"theme.base.removed"`   // used on removed lines
+	Unchanged color.Color `defaultFrom:"theme.base.unchanged"` // used on unchanged lines
```

Comment:
I just had a thought; `kubectl diff` outputs a YAML diff. What if the `DiffPrinter` parsed the output and handed over the output without the prefix `+`/`-` to the `YAMLPrinter`?

Doing the following already shows a quite promising coloring:

```bash
kubectl diff -f some-file.yaml | kubectl get -o yaml --kubecolor-stdin
```

(The `--kubecolor-stdin` flag is a special one that makes it only render the output instead of actually executing the command)

Then the colors here could be adding background colors to the output instead of foreground/text colors.

This should probably be put into a future enhancement issue as that would explode the scope of this PR.